packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,55 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (this.match(tt._with)) {
```

Comment:
To implement the `[no LineTerminator]` here, you can add this:
```suggestion
    if (this.match(tt._with) && !this.isLineTerminator()) {
```

Reply:
Do we need to add this?

Reply:
Yes, as mentioned at https://github.com/babel/babel/pull/10962#discussion_r391979487

Reply:
@nicolo-ribaudo Added this check as well and the tests are also updated. Can you review them once and see if that is the error that we expect?

Reply:
nit: It suffices to check `!this.hasPrecedingLineBreak()` because it is guarded by `this.match(tt._with)`.